main.go

```diff
@@ -0,0 +1,192 @@
+		Name:              "patas bravas",
+		Ingredients:       []string{"potato", "tomato", "olive oil", "onion", "garlic", "paprika"},
+		PrepTimeInMinutes: 80,
+	}
+
+	friedRiceRecipe := Recipe{
+		Name:              "fried rice",
+		Ingredients:       []string{"rice", "soy sauce", "egg", "onion", "pea", "carrot", "sesame oil"},
+		PrepTimeInMinutes: 40,
+	}
+
+	// Create an interface of all the created recipes
+	recipes := []interface{}{eloteRecipe, locoMocoRecipe, patatasBravasRecipe, friedRiceRecipe}
+	insertManyResult, err := collection.InsertMany(context.TODO(), recipes)
+	if err != nil {
+		fmt.Println("Something went wrong trying to insert the new documents:")
+		panic(err)
+	}
+	fmt.Println("the following documents were inserted:", insertManyResult.InsertedIDs)
+
+	/*
+	 * *** FIND DOCUMENTS ***
+	 *
+	 * Now that we have data in Atlas, we can read it. To retrieve all of
+	 * the data in a collection, we create a filter for recipes that take
+	 * less than 45 minutes to prepare and sort by name (ascending)
+	 */
+
+	var filter = bson.M{"prepTimeInMinutes": bson.M{"$lt": 45}}
+	options := options.Find()
+
+	// Sort by `name` field ascending
+	options.SetSort(bson.D{{"name", 1}})
+
+	cursor, err := collection.Find(context.TODO(), filter, options)
+	if err != nil {
+		fmt.Println("Something went wrong trying to find the documents:")
+		panic(err)
+	}
+
+	defer func() {
+		cursor.Close(context.Background())
+	}()
+
+	// Loop through the returned recipes
+	for cursor.Next(ctx) {
+		recipe := Recipe{}
+		err := cursor.Decode(&recipe)
+
+		// If there is an error decoding the cursor into a Recipe
+		if err != nil {
+			fmt.Println("cursor.Next() error:")
+			panic(err)
+		} else {
+			fmt.Println(recipe.Name, "has", len(recipe.Ingredients), "ingredients, and takes", recipe.PrepTimeInMinutes, "minutes to make.")
+		}
+	}
+
+	// We can also find a single document. Let's find the first document
+	// that has the string "fried rice" in the name.
+	var result Recipe
+	var myFilter = bson.D{{"name", "fried rice"}}
+	e := collection.FindOne(context.TODO(), myFilter).Decode(&result)
+	if e != nil {
+		fmt.Println("Something went wrong trying to find one document:")
+		panic(e)
+	}
+	fmt.Println("found a document with the name fried rice", result)
+
+	/*
+	 * *** UPDATE A DOCUMENT ***
+	 *
+	 * You can update a single document or multiple documents in a single call.
+	 *
+	 * Here we update the prepTimeInMinutes value on the document we
+	 * just found.
+	 */
+
+	var updateDoc = bson.D{{"$set", bson.D{{"prepTimeInMinutes", 72}}}}
+	var myRes = collection.FindOneAndUpdate(ctx, myFilter, updateDoc, nil)
+	if myRes.Err() != nil {
+		fmt.Println("Something went wrong trying to update one document:")
+		panic(myRes.Err())
+	}
+	doc := bson.M{}
+	decodeErr := myRes.Decode(&doc)
+	if decodeErr != nil {
+		fmt.Println("Something went wrong trying to decode the document:")
+		panic(decodeErr)
+	}
+	fmt.Println("Here is the updated document:", doc)
```

Comment:
I wonder if we should format this a little more cleanly and hide the ObjectID for reasons similar to described in my previous comment. Currently, this is the output:

Here is the updated document: map[_id:ObjectID("5f0dc2d91083054c1b3031ef") ingredients:[rice soy sauce egg onion pea carrot sesame oil] name:fried rice prepTimeInMinutes:40]

Perhaps the output of the other examples could be cleaned up as well if they aren't already.

Reply:
We could decode it into a Recipe{} struct to hide the _id, and then pretty print it using `json.MarshalIndent`. 

The code would be:
```
	_recipe := Recipe{}
	decodeErr := myRes.Decode(&_recipe)

	updatedRecipe, _ := json.MarshalIndent(_recipe, "", "\t")
	fmt.Println("The following document has been updated: \n", string(updatedRecipe))
``` 

and this would output:

```
The following document has been updated: 
 {
	"Name": "patas bravas",
	"Ingredients": [
		"potato",
		"tomato",
		"olive oil",
		"onion",
		"garlic",
		"paprika"
	],
	"prepTimeInMinutes": 72
}

```